weaviate/graphql/nearmultivectorbuilder.go

```diff
@@ -0,0 +1,147 @@
+)
+
+type NearMultiVectorArgumentBuilder struct {
+	certainty         float32
+	distance          float32
+	targetCombination dto.TargetCombination
+	targetVectors     []string
+	vectorPerTarget   map[string][]float32
+	withCertainty     bool
+	withDistance      bool
+}
+type targets struct {
+	combinationMethod string
+	targetVectors     []string
+	weights           map[string]float32
+}
+
+func (m *NearMultiVectorArgumentBuilder) toTargets() *targets {
+	combinationMethod := ""
+	switch m.targetCombination.Type {
+	case dto.Sum:
+		combinationMethod = "sum"
+	case dto.Average:
+		combinationMethod = "average"
+	case dto.Minimum:
+		combinationMethod = "minimum"
+	case dto.ManualWeights:
+		combinationMethod = "manualWeights"
+	case dto.RelativeScore:
+		combinationMethod = "relativeScore"
+	}
+	return &targets{
+		combinationMethod: combinationMethod,
+		targetVectors:     m.targetVectors,
+		weights:           m.targetCombination.Weights,
+	}
+}
+
+func (m *NearMultiVectorArgumentBuilder) Sum(targetVectors ...string) *NearMultiVectorArgumentBuilder {
+	if len(targetVectors) > 0 {
+		m.targetVectors = targetVectors
+		m.targetCombination = dto.TargetCombination{Type: dto.Sum}
+	}
+	return m
+}
+
+func (m *NearMultiVectorArgumentBuilder) Average(targetVectors ...string) *NearMultiVectorArgumentBuilder {
+	if len(targetVectors) > 0 {
+		m.targetVectors = targetVectors
+		m.targetCombination = dto.TargetCombination{Type: dto.Average}
+	}
+	return m
+}
+
+func (m *NearMultiVectorArgumentBuilder) Min(targetVectors ...string) *NearMultiVectorArgumentBuilder {
+	if len(targetVectors) > 0 {
+		m.targetVectors = targetVectors
+		m.targetCombination = dto.TargetCombination{Type: dto.Minimum}
+	}
+	return m
+}
+
+func (m *NearMultiVectorArgumentBuilder) ManualWeights(targetVectors map[string]float32) *NearMultiVectorArgumentBuilder {
+	if len(targetVectors) > 0 {
+		targetVectorsTmp := make([]string, 0, len(targetVectors))
+		for k := range targetVectors {
+			targetVectorsTmp = append(targetVectorsTmp, k)
+		}
+		m.targetVectors = targetVectorsTmp
+		m.targetCombination = dto.TargetCombination{Type: dto.ManualWeights, Weights: targetVectors}
+	}
+	return m
+}
+
+func (m *NearMultiVectorArgumentBuilder) RelativeScore(targetVectors map[string]float32) *NearMultiVectorArgumentBuilder {
+	if len(targetVectors) > 0 {
+		targetVectorsTmp := make([]string, 0, len(targetVectors))
+		for k := range targetVectors {
+			targetVectorsTmp = append(targetVectorsTmp, k)
+		}
+		m.targetVectors = targetVectorsTmp
+		m.targetCombination = dto.TargetCombination{Type: dto.RelativeScore, Weights: targetVectors}
+	}
+	return m
+}
+
+func (m *NearMultiVectorArgumentBuilder) WithVectorPerTarget(vectorPerTarget map[string][]float32) *NearMultiVectorArgumentBuilder {
+	if len(vectorPerTarget) > 0 {
+		m.vectorPerTarget = vectorPerTarget
+	}
+	return m
+}
+
+func (m *NearMultiVectorArgumentBuilder) WithVector(vector []float32) *NearMultiVectorArgumentBuilder {
```

Comment:
did you remove the "two-stage" builder on purpose? :) 

I added the them so that users at first only have the choice of choosing a combination type and then HAVE to chose a vector-type. Was more playing around and am not sure if that was a good idea, but would like to discuss it :)

Reply:
Yes, I removed it so that users have the option of not specifying the combination type thereby resorting to the default. This was to bring it in-line with the other clients, e.g. py-v4 doesn't require a method, right?
However, I'm not sold on it, do you think it would be useful to enforce a choice from the user?

Reply:
>  However, I'm not sold on it, do you think it would be useful to enforce a choice from the user?

Would be more about guiding users through the choices. They HAVE to add a vector (either single vector or vector per target), but you are right, the combination method has a default.

Let's leave it like it is